### PR TITLE
Add GRMHD prims after rollback mutator

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

I also added a few `ASSERT`s that were useful when debugging an extremely bizarre test failure.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.hpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/ActiveGrid.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.hpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#include "Evolution/DgSubcell/Tags/DidRollback.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.hpp]: forward declare?

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    const EquationsOfState::EquationOfState<true, ThermodynamicDim>&
+        eos) noexcept {
+  if (did_rollback) {
+    const size_t num_grid_points = subcell_mesh.number_of_grid_points();
+    ASSERT(prim_vars->number_of_grid_points() != num_grid_points,
+           "Doing rollback and the prim_vars should not be the size of the "
+           "subcell mesh");
+    // Project a copy of the pressure from the DG grid to the FD grid to give
+    // high-order initial guess for prim recovery.
+    const Scalar<DataVector> fd_pressure =
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.cpp]: isn't this the dg pressure?

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.cpp [hunk 4]
```diff
@@ -0,0 +1,112 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.hpp"
+
+#include <algorithm>
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimsAfterRollback.cpp]: unused?

File: tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_PrimsAfterRollback.cpp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  for (size_t i = 0; i < 3; ++i) {
+    inv_spatial_metric.get(i, i) = 1.0 / spatial_metric.get(i, i);
+  }
+  const Scalar<DataVector> sqrt_det_spatial_metric{
+      sqrt(get(determinant(spatial_metric)))};
+
+  std::unique_ptr<EquationsOfState::EquationOfState<true, 1>> eos =
+      std::make_unique<EquationsOfState::PolytropicFluid<true>>(1.4, 5.0 / 3.0);
+
+  // Compute the conservatives on the DG grid by first computing the primitives
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_PrimsAfterRollback.cpp]: aren't the following computations done on the subcell grid?

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Review by fmahebert — COMMENTED** [review]
I have no additional comments

**Comment by nilsdeppe** [comment]
I've rebased and squashed in the changes. Thanks for the reviews! :)

**Review by fmahebert — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**